src/map/hooks.es

```diff
@@ -0,0 +1,86 @@
+import { useEffect, useState, useRef } from 'react';
+import ReactMapboxGl from 'react-mapbox-gl';
+import { NavigationControl } from 'mapbox-gl';
+
+import { getMedia, media } from '../utils';
+import { getBoundsCenter, fitBounds } from './utils';
+
+
+export const useMapboxComponent = (locked, lockedMobile, noAttribution) => {
+  const [, setState] = useState();
+  const ref = useRef(null);
+
+  useEffect(() => {
+    const isMobile = !getMedia(global, media.mediaM);
+    const interactive = isMobile ? !lockedMobile : !locked;
+
+    ref.current = ReactMapboxGl({
+      dragPan: interactive,
+      keyboard: false,
+      doubleClickZoom: false,
+      scrollZoom: false,
+      injectCSS: false,
+      dragRotate: false,
+      pitchWithRotate: false,
+      apiUrl: null,
+      attributionControl: !noAttribution,
+    });
+
+    // Force re-render
+    setState({});
```

Comment:
this is done to update component after ref is set.
With isReady state it will be rendered only one time after first initialization.
We need to be able to recreate this component after some props were changed